src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -876,7 +933,10 @@ class Peer(
                 val balance = cmd.wallet.confirmedBalance
                 when (val channel = channels.values.firstOrNull { it is Normal }) {
                     is ChannelStateWithCommitments -> {
-                        val feerate = onChainFeeratesFlow.filterNotNull().first().fundingFeerate
+                        val targetFeerate = onChainFeeratesFlow.filterNotNull().first().fundingFeerate
+                        val weight = FundingContributions.computeWeightPaid(isInitiator = true, commitment = channel.commitments.active.first(), walletInputs = cmd.wallet.confirmedUtxos, localOutputs = emptyList())
+                        val (feerate, _) = watcher.client.computeSpliceCpfpFeerate(channel.commitments, targetFeerate, spliceWeight = weight, logger)
+
                         logger.info { "requesting splice-in using confirmed balance=$balance feerate=$feerate" }
 
                         val feeEstimate = Transactions.weight2fee(feerate, 610 + cmd.wallet.confirmedUtxos.size * Transactions.p2wpkhInputWeight)
```

Comment:
We should use `weight` here instead of this hard-coded value.

Reply:
We should use `weight` instead of `610 + cmd.wallet.confirmedUtxos.size * Transactions.p2wpkhInputWeight` right?

Reply:
Exactly, and we probably don't even need this `feeEstimate`: can't we use the fee provided by `computeSpliceCpfpFeerate`?

Reply:
Indeed, I noticed that too. It's now much cleaner in c4d3cd6d.